vunit/sim_if/xcelium.py

```diff
@@ -0,0 +1,694 @@
+            for library in self._libraries:
+                args += ['-reflib "%s"' % library.directory]
+            args += ['-input "@set intovf_severity_level ignore"']
+            if config.sim_options.get("disable_ieee_warnings", False):
+                args += [
+                    '-input "@set pack_assert_off { std_logic_arith numeric_std }"'
+                ]
+            args += [
+                '-input "@set assert_stop_level %s"' % config.vhdl_assert_stop_level
+            ]
+            if launch_gui:
+                args += ["-access +rwc"]
+                args += ['-linedebug']
+                args += ['-classlinedebug']
+                if use_indago:
+                    args += ["-debug_opts indago_interactive"]
+                    if step == "elaborate":
+                        args += ['-persistent_sources_debug']
+                    else:
+                        args += ["-nolock"]
+                else:
+                    # Use simvision
+                    args += ["-gui"]
+
+                if step == "elaborate":
+                    pass
+                    #args += ["-uvmlinedebug"]
+
+                gui_options = config.sim_options.get("xcelium.xrun_sim_flags.gui", [])
+                if gui_options is not None:
+                    args = gui_options + args
+                else:
+                    args += ['-input "@run"']
+            else:
+                args += ["-access +r"]
+                args += ['-input "@run"']
+
+            if config.architecture_name is None:
+                # we have a SystemVerilog toplevel:
+                args += ["-top %s.%s:sv" % (config.library_name, config.entity_name)]
+            else:
+                # we have a VHDL toplevel:
+                args += [
+                    "-top %s.%s:%s"
+                    % (
+                        config.library_name,
+                        config.entity_name,
+                        config.architecture_name,
+                    )
+                ]
+            argsfile = "%s/xrun_%s.args" % (script_path, step)
+            write_file(argsfile, "\n".join(args))
+            if not run_command(
+                [cmd, "-F", relpath(argsfile, script_path)],
+                cwd=script_path,
+                env=self.get_env(),
+            ):
+                return False
+
+        return True
+
+    def _hdlvar_args(self):
+        """
+        Return hdlvar argument if available
+        """
+        if self._hdlvar is None:
+            return []
+        return ['-hdlvar "%s"' % self._hdlvar]
+
+    @staticmethod
+    def _generic_args(entity_name, generics):
+        """
+        Create xrun arguments for generics/parameters
+        """
+        args = []
+        for name, value in generics.items():
+            if _generic_needs_quoting(value):
+                args += ['''-gpg "%s.%s => \\"%s\\""''' % (entity_name, name, value)]
```

Comment:
First, this seems appropriate for list comprehension. Second, why are there tripple `'''`?? This file has been using `"` for most quotes so this was awkward at first to read.

Reply:
I think this code was originally mine (back when f-strings were too new), and the triple single-quotes make this more readable IMO - it's a string with quotes that contains escaped quotes.
I agree with the use of a list comprehension and would also use f-strings now

Reply:
That's fair. I would honestly run this file through [black](https://black.readthedocs.io/en/stable/) with preview features enabled so it reformats strings and see what it comes up with.